test/WebSites/ModelBindingWebSite/Controllers/VehicleController.cs

```diff
@@ -37,7 +37,7 @@ public class VehicleController : Controller
         [HttpPut("/api/vehicles/{id}")]
         [Produces("application/json")]
         public object UpdateVehicleApi(
-            [Range(1, 500)] int id,
```

Comment:
This started working by coincidence when we added `GetMetadataForParameter`. We never really talked about this or intended it :laughing: @harshgMSFT is currently hard at work removing `GetMetadataForParameter`

MVC never had support for this. 

Yes, `DataAnnotations` attributes can be placed on parameters, but MVC and WebAPI have never supported this - they are only supported on model property and types.

Try out the following in MVC 4
```
public class HomeController : Controller
    {
        //
        // GET: /Home/

        public ActionResult Index([Range(0, 5)] int id)
        {
            return View();
        }

        public ActionResult WithModel(Model model)
        {
            return View();
        }

        public class Model
        {
            [Range(0, 5)]
            public int Id { get; set; }
        }
    }
```